go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -142,6 +143,48 @@ func (c *Controller) Run() error {
 	return nil
 }
 
+// initializeNamespaceTracker populates the namespace-tracker with NAD namespaces who owned by the controller.
+func (c *Controller) initializeNamespaceTracker() error {
+	cudns, err := c.cudnLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	if len(cudns) == 0 {
+		return nil
+	}
+
+	nads, err := c.nadLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	if len(nads) == 0 {
+		return nil
+	}
+	indexedNADs := map[string]netv1.NetworkAttachmentDefinition{}
+	for _, nad := range nads {
+		if nad != nil {
+			indexedNADs[nad.Namespace+"/"+nad.Name] = *nad.DeepCopy()
+		}
+	}
+
+	for _, cudn := range cudns {
+		cudnCopy := cudn.DeepCopy()
+		for nadKey, nad := range indexedNADs {
```

Comment:
I think you may not need indexedNADs, and just iterate over `nads` from the `List` call

Reply:
Having a map enable optimize this loop in a way that it wont process a NAD that already been tracked.
When a NAD is tracked it is removed from the map in line 181.

Reply:
oh I completely missed that part, I don't think that it is a good idea to modify the map you are iterating over inside the loop.

Reply:
somewhat of a grey area https://groups.google.com/g/golang-nuts/c/eJqwQLONhLs but probably should work

Reply:
Added comment to clearfy when we remove iterated entries.